File: modules/Widgets/SubClassing.ts
import { Widget, Gtk } from "imports"

const Switch = Widget.subclass(Gtk.Switch, "AgsSwitch");
const TextView = Widget.subclass(Gtk.TextView, "AgsTextView");
const ComboBoxText = Widget.subclass(Gtk.ComboBoxText, "AgsComboBoxText");
const ComboBox = Widget.subclass(Gtk.ComboBox, "AgsComboBox");

export {
  Switch,
  TextView,
  ComboBox,
  ComboBoxText,
};

File: Rust/Modules/Lib/SubClassing.rs
use imports::{Widget, Gtk};

let Switch = Widget::subclass::<Gtk::Switch>("AgsSwitch");
let TextView = Widget::subclass::<Gtk::TextView>("AgsTextView");
let ComboBoxText = Widget::subclass::<Gtk::ComboBoxText>("AgsComboBoxText");
let ComboBox = Widget::subclass::<Gtk::ComboBox>("AgsComboBox");

pub {
    Switch,
    TextView,
    ComboBox,
    ComboBoxText,
};
